### PR TITLE
Enhance ResGen tool to support a command line argument

### PR DESCRIPTION
**Pull request by iSazonov**
Related #3400

Motivation
---
ResGen tool process all resx files in PowerShell Repo at a time. It is not problem because it works very fast. But updating multiple cs files causes all Repo projects to be recompiled. It really slows down.

We still can not use native resgen from dotnet-cli. See https://github.com/PowerShell/PowerShell/issues/2882#issuecomment-304300044
So we should enhance our ResGen tool, use MSBuild capabilities and process resx files one at a time to avoid unnecessary rebuilds.

Fix
---
We can specify a resx file path parameter on the command line and process resx files one at a time.

Follow-Up work
---
We will clean up Build.psm1 and modify csproj files to use the new feature of ResGen tool with MSBuild for faster project build.

### CODE REVIEW

**Comment by vors** [comment]
I think it's a good idea.

> We will clean up Build.psm1 and modify csproj files to use the new feature of ResGen tool for faster project build.

Couple comments: I don't think we should try to be smart and auto-detect set of files for resgen in build.psm1. We should just figure out the way to switch to msbuild resgen sooner or later.
Let's just expose this functionality thru `Start-ResGen` optional parameter in the same PR.

**Comment by iSazonov** [comment]
Yes, it is all for MSBuild - I updated the comment.

**Comment by vors** [comment]
oh, do you mean that you know how to plug it into csproj files so it's called automatically?

**Comment by iSazonov** [comment]
Yes. You can see sample in #3870

**Review by TravisEz13 — APPROVED** [review]

**Comment by iSazonov** [comment]
@TravisEz13 Is the PR ready to merge?

File: src/ResGen/Program.cs [hunk 1]
```diff
@@ -10,12 +10,30 @@ public class Program
...
-            // so we can use relative paths
-            foreach (string folder in Directory.EnumerateDirectories(".."))
+            IEnumerable<string> dirs;
+            string fileFilter;
+
+            if (args.Length == 1)
+            {
+                // We are assuming resgen is run with 'dotnet run pathToResxFile.resx'.
+                fileFilter = Path.GetFileName(args[0]);
+                string moduleDirectory = Directory.GetParent(Path.GetDirectoryName(args[0])).FullName;
```
Comment (daxian-dbw) [inline src/ResGen/Program.cs]: > string moduleDirectory = Directory.GetParent(Path.GetDirectoryName(args[0])).FullName;
```
PS:39> [System.IO.Directory]::GetParent

OverloadDefinitions
-------------------
static System.IO.DirectoryInfo GetParent(string path)
```
A `DirectoryInfo` object cannot be converted to a string ...

Comment (daxian-dbw) [inline src/ResGen/Program.cs]: Why not `Path.GetDirectoryName(Path.GetDirectoryName(args[0]))`?

**Review by daxian-dbw — APPROVED** [review]

**Comment by daxian-dbw** [comment]
LGTM except for a minor comment.

**Review by vors — APPROVED** [review]

File: src/ResGen/Program.cs (hunk 1 above)
Comment (iSazonov) [inline src/ResGen/Program.cs]: Good catch! I'll fix on next week.

Comment (iSazonov) [inline src/ResGen/Program.cs]: Done.

**Pull request merged**